### PR TITLE
Toponaming: Add missing TNP element name code; rough in more TNP tests

### PR DESCRIPTION
**Pull request by bgbsww**

### CODE REVIEW

**Comment by shaise** [comment]
@bgbsww ,

This totaly breaks the workbench - all properties of the object are gone:

before the fix:
image

after the fix:
image

**Comment by shaise** [comment]
-deleted wrong message-

**Comment by shaise** [comment]
@bgbsww ,

Indeed now the returned names are simple names.
However, when I open a file created prior to TNP, the model breaks when I recompute or change, since all original edges are remaped to new ones. 
Isn't there a way to make TNP backward compatible to pre-TNP files?
See attached. 

[Error_created_pre_tnp.zip](https://github.com/user-attachments/files/15748998/Error_created_pre_tnp.zip)

**Comment by bgbsww** [comment]
Thanks, I wonder why all the tests passed.  Guess we'll need a new test to catch this case.  Anyway, this sounds like a failure to reassign / recalculate on load, I'll take a look at that.

**Comment by shaise** [comment]
@bgbsww , sorry for disturbing you so much. Here is another case when the name is not simple. When doing multi selection, only the first selection is simple, all others are TNP:

https://github.com/FreeCAD/FreeCAD/assets/5813568/c512a195-faa8-4bc3-b641-5800a137a95a

**Comment by bgbsww** [comment]
You are definitely not disturbing me; you've triggered a mini avalanche of problems to look into.  The Data::findElementName method at the base of all of this is called 49 times in LS3, but only 35 in main; there are 12 source files to compare and clean up, including two in Selection which underpin pretty much everything.

So excellent find, I'll be busy for a bit...

File: tests/src/App/ElementNamingUtils.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include <gtest/gtest.h>
+
+#include <App/Application.h>
+#include <App/ElementMap.h>
+#include <src/App/InitApplication.h>
+
+// NOLINTBEGIN(readability-magic-numbers)
```
Comment (chennes) [inline tests/src/App/ElementNamingUtils.cpp:9]: Missing NOLINTEND

**Pull request merged**